### PR TITLE
Add install script for the Bluetooth monitor script

### PR DESCRIPTION
**Pull request by Landrash**
## Description:
Adds install script for the Bluetooth [Montitor](https://github.com/andrewjfreyer/monitor) script by [andrewjfreyer](https://github.com/andrewjfreyer)

**  Depends on changes in #201 **

## Checklist (Required):
  - [x] The code change is tested and works locally.
  - [x] The code is compliant with [Contributing guidelines](https://github.com/home-assistant/hassbian-scripts/blob/master/.github/CONTRIBUTING.md)

### If pertinent:
  - [x] Script has validation check of the job.
  - [x] Created/Updated documentation at `/docs`

### CODE REVIEW

File: package/opt/hassbian/suites/monitor.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+function monitor-install-package {
+echo -n "Installing Mosquitto package: "
+
+echo "Checking the installation..."
+validation=$(pgrep -f mosquitto)
+if [ ! -z "${validation}" ]; then
+  echo "Existing Mosquitto install detected. Skipping install of Mosquitto."
+else
+  echo "Installing Mosquitto"
+sudo hassbian-config install mosquitto
```
Comment (ludeeus) [inline package/opt/hassbian/suites/monitor.sh]: `sudo` is not needed inside the scripts :)

Should there also be an implementation of `--accept` here so the script can be run without user input?

File: package/opt/hassbian/suites/monitor.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+echo "Creating Monitor folder..."
+mkdir /opt/monitor || exit
+cd /opt || exit
+
+echo "Cloning Monitor git repository"
+git clone git://github.com/andrewjfreyer/monitor
+
+echo "Running interactive setup"
+cd /opt/monitor || exit
+sudo bash /opt/monitor/monitor.sh
```
Comment (ludeeus) [inline package/opt/hassbian/suites/monitor.sh]: `sudo` is not needed inside the scripts :)

**Review by ludeeus — COMMENTED** [review]
Added some comments, I will probably have some time later today to test it :)

Since we now have `remove` option in the tool, maybe add a remove(cleanup) function too? :)
and maybe an `upgrade` function to fetch newest files from git?

File: package/opt/hassbian/suites/monitor.sh (hunk 1 above)
Comment (Landrash) [reply package/opt/hassbian/suites/monitor.sh]: True on both account. Changing.

File: package/opt/hassbian/suites/monitor.sh (hunk 2 above)
Comment (Landrash) [reply package/opt/hassbian/suites/monitor.sh]: Haha, didn't consider that..

**Comment by ludeeus** [comment]
I got some errors during the install:
```
Error: Required package 'hcidump' not found. Please install 'bluez-hcidump' (e.g                                                                             ., sudo apt-get install bluez-hcidump).
Error: Required package 'bc' not found. Please install 'bc' (e.g., apt-get install bc)
```
And when I tried starting the service it fails:
```
● monitor.service - Monitor Service
   Loaded: loaded (/etc/systemd/system/monitor.service; enabled; vendor preset: enabled)
   Active: activating (auto-restart) (Result: exit-code) since Sun 2018-10-14 19:18:36 UTC; 3s ago
  Process: 10060 ExecStart=/bin/bash /opt/monitor/monitor.sh & (code=exited, status=1/FAILURE)
 Main PID: 10060 (code=exited, status=1/FAILURE)

Oct 14 19:18:36 hassbian systemd[1]: monitor.service: Failed with result 'exit-code'.
```

It _may_ be limited to my system, will try again with a fresh copy

**Comment by Landrash** [comment]
This is my fault. I forgot some upstream cleanups I did with dependencies
in another branch.. fixing later this week

**Comment by Landrash** [comment]
> I got some errors during the install:
> 
> ```
> Error: Required package 'hcidump' not found. Please install 'bluez-hcidump' (e.g                                                                             ., sudo apt-get install bluez-hcidump).
> Error: Required package 'bc' not found. Please install 'bc' (e.g., apt-get install bc)
> ```
> And when I tried starting the service it fails:
> 
> ```
> ● monitor.service - Monitor Service
>    Loaded: loaded (/etc/systemd/system/monitor.service; enabled; vendor preset: enabled)
>    Active: activating (auto-restart) (Result: exit-code) since Sun 2018-10-14 19:18:36 UTC; 3s ago
>   Process: 10060 ExecStart=/bin/bash /opt/monitor/monitor.sh & (code=exited, status=1/FAILURE)
>  Main PID: 10060 (code=exited, status=1/FAILURE)
> 
> Oct 14 19:18:36 hassbian systemd[1]: monitor.service: Failed with result 'exit-code'.
> ```
> It _may_ be limited to my system, will try again with a fresh copy

Should work now by running with script or service. 
Dependency changes are in #201 to address problems with general bluetooth components sine I did cleanups in that branch.

**Comment by Landrash** [comment]
@ludeeus I'll wait with removal and update until the upstream project gets a bit more stable since it currently is undergoing quite a few changes.

File: package/opt/hassbian/suites/monitor.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+echo "Creating Monitor folder..."
+mkdir /opt/monitor || exit
+cd /opt || exit
+
+echo "Cloning Monitor git repository"
+git clone git://github.com/andrewjfreyer/monitor
+
+echo "Running interactive setup"
+cd /opt/monitor || exit
+chmod /opt/monitor/monitor.sh || exit
```
Comment (ludeeus) [inline package/opt/hassbian/suites/monitor.sh]:
```suggestion
chmod +x /opt/monitor/monitor.sh || exit
```
The script fails here, guessing the idea was to make it executable?

```
chmod: missing operand after ‘/opt/monitor/monitor.sh’
```

**Review by ludeeus — CHANGES_REQUESTED** [review]
Tested this again after testing #201 and it works much better now 👍 
I have added 1 suggestion :)

**Review by ludeeus — APPROVED** [review]
🐬

**Pull request merged**